util/rate_limiter_test.cc

```diff
@@ -416,6 +416,84 @@ TEST_F(RateLimiterTest, LimitChangeTest) {
   ROCKSDB_NAMESPACE::SyncPoint::GetInstance()->DisableProcessing();
 }
 
+TEST_F(RateLimiterTest, AvailableByteSizeExhaustTest) {
```

Comment:
I think the UT has test correctness well but we could see whether we can make it simpler in the following ways:
(1) To ensure the timing of 3 requests, do we have to use sync point especially for the first one? It's also not super obvious/straightforward to see how the test controls the timing of the 3 requests to ensure the following cases don't happen
(a) 500 request arrives on the second refill (so available bytes has 400+500 and 500 request accidentally early returns)
(b) 300 request arrives when 500 request finishes/another refill comes

Can we improve on that? Not sure if existing UTs have encountered similar issue with yours?

(2) I felt it's still subtle to understand how the UT can fail on code without your PR or an incorrectly implemented PR. Can we make it more straightforward so that it's easier to see the A/B effect of your test?

Reply:
Still working on the test, but just wanted to make one thing clear that `available_bytes_` should never be greater than 500 (in this test case) since we are now setting `available_bytes_ = refill_bytes_per_period;`

Reply:
Yes.